src/AstAnalyser.js

```diff
@@ -14,10 +15,25 @@ import { JsSourceParser } from "./JsSourceParser.js";
 export class AstAnalyser {
   /**
    * @constructor
-   * @param { SourceParser } [parser]
+   * @param options
    */
-  constructor(parser = new JsSourceParser()) {
-    this.parser = parser;
+  constructor(options = {}) {
+    if (options.customParser !== undefined) {
+      assert(options.customParser instanceof JsSourceParser || typeof options.customParser === "object",
+        `customParser must be an instance of JsSourceParser or an object`);
+    }
+    if (options.customProbe !== undefined) {
+      assert(Array.isArray(options.customProbe), `customProbe must be an array`);
+    }
+    if (options.isReplacing !== undefined) {
+      assert(typeof options.isReplacing === "boolean", `isReplacing must be a boolean`);
```

Comment:
I'm not sure for these `assert` because it throws `AssertionError`. Usually we wanna throw `TypeError` instead.

Reply:
I still think we should get rid of these type tests, wdyt?
Checking for `undefined` value in order to set default values is enough if we don't use TS.

Reply:
Yep, it should be ok

Reply:
The assert part has been removed, typing and default value are enough agreeed